src/buildstream/_platform/platform.py

```diff
@@ -44,16 +44,16 @@ def _setup_sandbox(self):
         # Try to setup buildbox-run sandbox, otherwise fallback to the dummy sandbox.
         try:
             self._check_sandbox(SandboxBuildBoxRun)
-        except (SandboxError, utils.ProgramNotFoundError):
+        except (SandboxUnsupportedError, utils.ProgramNotFoundError):
             pass
 
     def _check_sandbox(self, Sandbox):
         Sandbox._dummy_reasons = []
         try:
             Sandbox.check_available()
-        except SandboxError as Error:
+        except SandboxUnsupportedError as Error:
```

Comment:
Let’s fix the `Error` variable and call it `e` if we still need it after correcting the `raise` statement below

Reply:
Also in the line below we are collecting this weird reasons list of strings, it appears to be the last place justifying the existence of this `Sandbox._dummy_reasons` variable, any chance we can take this opportunity to get rid of it ? (Not sure we need this similar variable `Platform.dummy_reasons` either).

Reply:
The reason I didn't modify it in #1436 is that I wanted to have it backportable to bst-1 (but turns out bst-1 didn't have a dummy sandbox at all, so whatever).

There are actually two different checks `Sandox.check_available()` and `Sandbox._check_sandbox_config()` (is there a reason why the former is "public"?) and we want to cache the result of the former. That's essentially why dummy_reasons exist.

What I would recommend is to do the caching in the `Sandbox` class itself, such that calling `_check_sandbox_config()` will raise the newly introduced `SandboxUnsupportedError` if buildbox-run isn't available before actually checking the config (and possibly drop `Sandox.check_available()` altogether if it's not needed for your new check?)

Reply:
Ok, updated based on that feedback. I renamed SandboxBuildBoxRun.check_available() to `setup()` as that's more correct based on what it does. I kept it public as, the whole _sandboxbuildboxrun module is private API, and setup() is called from outside the module. (I may be misunderstanding how the rules work in Bst though).

Reply:
It should be made private generally, so as not to take up any public namespace (In this instance it’s not very important given that the user does not subclass sandboxes).

One underscore for shared internal API, two underscores for module internal details (two underscores are only used in these public facing classes, where the distinction between “private” and “internal” is significant).

Reply:
This is renamed to _setup in 1cd18ff3c5b5c5c0d190bcfe36842f730c6a5ba0